infrastructure/ansible/roles/nginx/templates/etc_nginx_sites_enabled_default.j2

```diff
@@ -10,12 +10,26 @@ server {
     ssl_certificate_key /etc/letsencrypt/live/{{ inventory_hostname }}/privkey.pem; # managed by Certbot
 
     ssl on;
-    ssl_session_cache  builtin:1000  shared:SSL:10m;
     ssl_protocols  TLSv1.2 TLSv1.3;
     ssl_ciphers EECDH+AESGCM:EDH+AESGCM;
     ssl_prefer_server_ciphers on;
     access_log            /var/log/nginx/java_server.access.log;
 
+    ssl_prefer_server_ciphers on; 
+    ssl_dhparam /etc/nginx/dhparam.pem; # openssl dhparam -out /etc/nginx/dhparam.pem 4096
+    ssl_ecdh_curve secp384r1; # Requires nginx >= 1.1.0
+    ssl_session_timeout  10m;
+    ssl_session_cache shared:SSL:10m;
+    ssl_session_tickets off; # Requires nginx >= 1.5.9
+    ssl_stapling on; # Requires nginx >= 1.3.7
+    ssl_stapling_verify on; # Requires nginx => 1.3.7
+    resolver $DNS-IP-1 $DNS-IP-2 valid=300s;
+    resolver_timeout 5s; 
+    add_header Strict-Transport-Security "max-age=63072000; includeSubDomains; preload";
+    add_header X-Frame-Options DENY;
+    add_header X-Content-Type-Options nosniff;
+    add_header X-XSS-Protection "1; mode=block";
```

Comment:
This is good context. If we ever have a web browser user agent hit nginx, we would almost just as well reject the request. That would be a bit extreme as it might impair our efforts to do development and debug, but I think it speaks to our expectations/requirements, we really only expect TripleA app to be hitting nginx for the purpose of getting to data over http.